addons/package_hierarchy/models/stock_picking.py

```diff
@@ -24,18 +24,27 @@ def _compute_entire_package_ids(self):
     def _check_entire_pack(self):
         """Set u_result_parent_package_id when moving entire parent package."""
         super(StockPicking, self)._check_entire_pack()
+        self._set_u_result_parent_package_id()
+    
+    def _set_u_result_parent_package_id(self):
+        """Set u_result_parent_package_id when moving entire parent package."""
         for picking in self:
-            mls = picking.move_line_ids.filtered(lambda ml: not ml.u_result_parent_package_id)
+            mls = picking.move_line_ids.filtered(
+                lambda ml: not ml.u_result_parent_package_id
+            )
             # Get result packages of move lines without result parent package
             result_packages = mls.mapped('result_package_id')
             parent_packages = result_packages.mapped("package_id")
-            # get all children in bulk
+            # Get all children in bulk
             parent_packages.mapped('children_ids')
             for parent_package in parent_packages:
                 if len(parent_package.children_ids - result_packages) == 0:
                     pack_mls = mls.filtered(
-                        lambda ml: ml.result_package_id.package_id == parent_package and
-                                   not ml.u_result_parent_package_id)
-                    pack_mls.write({'u_result_parent_package_id': parent_package.id})
+                        lambda ml: (
```

Comment:
I would leave the style as it was.
Also style changes is better to have them in different commits, to distinguish code changes to style changes.